crocket/src/main/java/com/crocket/Model.java

```diff
@@ -152,16 +166,53 @@ public void updateAim(int degrees) {
         directionLine.setDegreeAngle(degrees);
     }
 
+    public void aimRight() {
+        directionLine.incrementDegreeAngle();
+    }
+
+    public void aimLeft() {
+        directionLine.decrementDegreeAngle();
+    }
+
     public void setPlayers(List<Player> players) {
         this.players = players;
+
+        activePlayer = players.get(0);
+
+        for (Player player : players) {
+            entities.add(player.getBall());
+            movables.add(player.getBall());
+        }
     }
 
     public void addPlayer(Player player) {
+        if (players == null) {
+            players = new ArrayList<Player>();
+        }
+
         players.add(player);
+
+        if (activePlayer == null) {
+            activePlayer = player;
+        }
+
+        entities.add(player.getBall());
+        movables.add(player.getBall());
     }
 
     public void clearPlayers() {
         players.clear();
```

Comment:
Might not be a problem but what would happen if we cleared and empty players list?

Reply:
Or does not have an instance

Reply:
I think it should work if the list is empty. Might be a problem if someone passes `null` in setPlayers(). I'll look into this